### PR TITLE
Glory

### PR DESCRIPTION
**Pull request by Glorycornel**
### Title

**Add Quiz Generator App with Next.js, TypeScript, and Tailwind CSS**

### Description

This pull request introduces a new Quiz Generator application built using Next.js with the App Router, TypeScript, and Tailwind CSS. The application allows users to create and manage quizzes and includes the following features:

- **Homepage:** Allows users to input quiz questions and answers.
- **Quiz Page:** Users can take the quiz and view their score.
- **Styling:** Implemented using Tailwind CSS for a responsive and modern design.
- **Client Components:** Utilizes Next.js's new client-side features for dynamic interactions.

### Changes

1. **Created a Next.js Application:**
   - Initialized the project using Next.js with TypeScript.
   - Used `pnpm` as the package manager for dependency management.

2. **Configured Tailwind CSS:**
   - Set up Tailwind CSS for styling with a custom `tailwind.config.js`.
   - Included Tailwind’s base, components, and utilities in `styles/globals.css`.
   
Screenshot from 2024-08-29 02-57-54

3. **Implemented Homepage (`src/app/page.tsx`):**
   - Added a form to input quiz questions and answers.
   - Enabled adding multiple questions dynamically.
   - Submitted data navigates to the Quiz Page with query parameters.

4. **Implemented Quiz Page (`src/app/quiz/page.tsx`):**
   - Displays the quiz questions retrieved from query parameters.
   - Handles user input and calculates a score.
   - Provides feedback and a button to return to the homepage.

5. **Updated File Structure:**
   - Organized the application files under the `src/app` directory following Next.js's App Router conventions.

6. **Added Client Component Directives:**
   - Marked components with `'use client';` to indicate client-side functionality.
   - 
Screenshot from 2024-08-29 03-02-08

### Code

- **`src/app/page.tsx`**: The homepage where users can create quizzes.
- **`src/app/quiz/page.tsx`**: The page where users can take the quiz and view their score.
- **`styles/globals.css`**: Includes global styles and Tailwind CSS imports.

### How to Test

1. **Install Dependencies:**
   ```bash
   pnpm install
   ```

2. **Run the Development Server:**
   ```bash
   pnpm dev
   ```

3. **Access the Application:**
   - Navigate to `http://localhost:3000` to view the homepage.
   - Add questions and submit to navigate to the quiz page.
   
Screenshot from 2024-08-29 03-00-43

4. **Verify Functionality:**
   - Ensure that you can add multiple questions and see them on the quiz page.
   - Check that the score is calculated and displayed correctly.

### Notes

- Ensure that the `next.config.js` and `tsconfig.json` are correctly configured.
- The app uses client-side components, and the use of `'use client';` is necessary for certain functionalities.

### Additional Information

- **Dependencies Added:** Next.js, Tailwind CSS, TypeScript.
- **Potential Improvements:** Could add unit tests, enhance error handling, or improve UX/UI based on feedback.

### CODE REVIEW

File: client/src/app/page.tsx.md
Comment (obinnaeye) [inline client/src/app/page.tsx.md:1]: This should be removed.

**Comment by obinnaeye** [comment]
Kindly add screenshots to the PR since this is frontend dev work.

File: client/src/app/page.tsx.md
Comment (Glorycornel) [reply client/src/app/page.tsx.md:1]: Ok, i'll do just that.

**Comment by Glorycornel** [comment]
All done @obinnaeye

**Review by julietowah — APPROVED** [review]
all good

**Pull request merged**